packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.ts

```diff
@@ -182,22 +183,43 @@ type UseShopperBasketsMutationArg = {
 }
 
 type ShopperBasketsClient = ApiClients['shopperBaskets']
-export type ShopperBasketsMutationType =
-    (typeof ShopperBasketsMutations)[keyof typeof ShopperBasketsMutations]
+export type ShopperBasketsMutationType = typeof ShopperBasketsMutations[keyof typeof ShopperBasketsMutations]
 
 /**
  * @private
  */
 export const getCacheUpdateMatrix = (customerId: string | null) => {
-    const updateBasketQuery = (basketId?: string) => {
+    const updateBasketQuery = (
+        basketId?: string,
+        response?: DataType<Client[ShopperBasketsMutationType]>
+    ) => {
         // TODO: we're missing headers, rawResponse -> not only {basketId}
         const arg = {basketId}
         return basketId
             ? {
                   update: [
                       {
                           name: 'basket',
-                          key: ['/baskets', basketId, arg]
+                          key: ['/baskets', basketId, arg],
+                          updater: () => response
+                      },
+                      {
+                          // Since we use baskets from custom basket query, we need to update it for any basket mutation
+                          name: 'customerBaskets',
+                          key: ['/customers', customerId, '/baskets', {customerId}],
+                          updater: (
+                              oldData: NonNullable<DataType<CustomerClient['getCustomerBaskets']>>
+                          ) => {
+                              const updatedBaskets = oldData.baskets?.map(
+                                  (basket: DataType<Client[ShopperBasketsMutationType]>) => {
+                                      return basket?.basketId === basketId ? response : basket
+                                  }
+                              )
+                              return {
+                                  ...oldData,
+                                  baskets: updatedBaskets
+                              }
+                          }
```

Comment:
General question. From the looks of it we are always updating the customers baskets cache, even if it doesn't find a match for the response. I imagine that the cache for the customer will automatically have some kind of last updated value set even tho the data itself hasn't changed.

I know this isn't a likely scenario, as in our case the response basket will always be in the customer basket, but there is a potential scenario where you wouldn't want this behaviour. If a customer service agent is adding products to a cart on behave of a customer, it's possible that the basket being updated isn't in the customers basket (aka the customer service agent). 

Anyway. Maybe we can only have the second update if we know that the response exists in the customer cache. Not a difficult change.